components/Layout/HomeHeader/HomeHeader.vue

```diff
@@ -63,14 +67,8 @@ export default Vue.extend({
         continue;
       }
 
-      const itemData = (
-        await this.$api.userLibrary.getItem({
-          userId: this.$auth.user?.Id,
-          itemId: id
-        })
-      ).data;
-
-      this.relatedItems[key] = itemData;
+      await this.fetchItem({ id });
```

Comment:
What about passing to this method a function as an argument? So we can pass it whatever Axios function and run it in the store instead of the component's context. That way we can query whole libraries or whatever we need instead of doing it item by item.

Also, we shouldn't await this so the component load isn't blocked.

Reply:
You can in the store then check if the response is an array or a single item and, if it's an array, add each of the inner objects.

You can do this easily with lodash

Reply:
I don't wanna use lodash too much but yeah :p

Reply:
> So we can pass it whatever Axios function and run it in the store instead of the component's context

I think a better way would be to make multiple actions for whatever we need. Most of the time, we're going to be touching this store from somewhere else anyway, so most of the API calls would be done elsewhere, I would think.

The Websocket update is really the only place where I can see us fetching things from the item store. All the other API calls would be triggered from somewhere else, then processed a bit to copy IDs as an array and `addOrUpdateItems` to the Items store.